### PR TITLE
Minor cleanups to FE_FaceQ.

### PR DESCRIPTION
**Pull request by bangerth**
This covers the requested changes in #4577.

### CODE REVIEW

File: source/fe/fe_face.cc [hunk 1]
```diff
@@ -353,8 +355,10 @@ std::vector<std::pair<unsigned int, unsigned int> >
 FE_FaceQ<dim, spacedim>::
 hp_quad_dof_identities (const FiniteElement<dim, spacedim> &fe_other) const
 {
+  Assert (dim >= 3, ExcInternalError());
+
   // this element is continuous only for the highest dimensional bounding object
-  if (dim !=3)
+  if (dim > 3)
     return
```
Comment (masterleinad) [inline source/fe/fe_face.cc]: This branch is (currently?) unreachable. Can't we just restrict the whole function to `dim==3`?

Comment (bangerth) [reply source/fe/fe_face.cc]: Yes, it's of course unreachable. I was trying to be conservative by making it truly dimension independent and in the process documenting (through the code) why `dim != 3` is not *logically* the correct condition.

We could remove the whole thing, but I think it serves an educational purpose.

**Comment by masterleinad** [comment]
You need to fix the failing test `hp/hp_quad_dof_identities_faceq.debug` by restricting it to `dim==3`.

**Comment by bangerth** [comment]
Fixed now!

**Comment by masterleinad** [comment]
Thanks!

**Pull request merged**